dev_docs/07_visual_testing_guide.md

```diff
@@ -0,0 +1,101 @@
+
+     ```javascript
+     import { renderComponent, takeSnapshot } from './visual.testUtils';
+     ```
+
+     You can also import and use the utility functions for managing component's visual states as needed. To get an overview of the different utility functions available, please check out [`visual.testUtils.js`](../jest.conf/visual.testUtils.js).
+
+2. **Write Tests**:
+   - Use the utility functions to render components and take snapshots. For example:
+
+     ```javascript
+     describe.visual('KButton Visual Tests', () => {
+       it('Sample test for KButton', async () => {
+         await renderComponent('KButton', { text: 'Raised Button', appearance: 'raised-button' });
+         await takeSnapshot('KButton - Raised Button', { widths: [375, 520] });
+       });
+     });
+     ```
+
+   - Make sure to use `describe.visual` or `it.visual` instead of the default notations for writing test blocks containing visual tests so as to prevent any unexpected behavior. These custom blocks add a [Visual] tag to the test name whose presence or absence are then checked using a regex pattern based on the type of tests executed. This implementation helps us to determine which test blocks should be executed by Jest and which ones should be skipped.
+
+3. **Simulate User Interactions**:
+   - Use the custom commands to simulate user interactions. For example, to simulate the *'click'* user event, you can do something like:
+
+     ```javascript
+      await click('button');
+     ```
+
+     Here, *'button'* is the CSS selector for the component. You can pass different selectors to the functions, exposed by [`visual.testUtils.js`](../jest.conf/visual.testUtils.js), to simulate user interaction as per requirement.
+
+## Implementation details
+
+  The visual testing mechanism uses the following dependencies to ensure components render correctly under various conditions:
+
+  - **Puppeteer:** for interacting with the testing environment and the rendered components.
+  - **Jest-Puppeteer:** to provide all required configuration to run tests using Puppeteer.
+  - **Percy:** to take snapshots for comparing visual diffs.
+  
+  The key parts of the mechanism include:
+
+1. **Configuration Files**: Since we are using Jest for both unit and visual tests, there are two separate configuration files for visual tests apart from the ones being used for unit tests so as to ensure separation of logic needed for running both types of tests.
+   - [***visual.index.js***](../jest.conf/visual.index.js): Configures Jest-Puppeteer and includes server checks to ensure the visual testing playground is up and running.
+   - [***visual.setup.js***](../jest.conf/visual.setup.js): Sets up global functions and constants needed for visual tests.
+
+2. **Utility Functions** [***(visual.testUtils.js)***](../jest.conf/visual.testUtils.js): We are also using a separate file that contains all the utility functions that are needed for writing visual tests.
```

Comment:
I think these would be better listed in the "Writing visual tests" section. Generally, I wouldn't expect that people who just want to write a visual test and understand how to do it will need to go to this section. The documentation would benefit from separating clearly these two perspectives.